gufe/components/explicitmoleculecomponent.py

```diff
@@ -0,0 +1,200 @@
+        elif (n_confs := len(conformers)) > 1:
+            warnings.warn(f"Molecule provided with {n_confs} conformers. "
+                          "Only the first will be used.")
+
+        self._rdkit = rdkit
+        self._hash = self._hashmol(name=name)
+
+    def _hashmol(self, name):
+        return hashmol(self._rdkit, name=name)
+
+    def to_rdkit(self) -> RDKitMol:
+        """Return an RDKit copied representation of this molecule"""
+        return Chem.Mol(self._rdkit)
+
+    @classmethod
+    def from_rdkit(cls, rdkit: RDKitMol, name: str = ""):
+        """Create a SmallMoleculeComponent, copying from an RDKit Mol"""
+        return cls(rdkit=Chem.Mol(rdkit), name=name)
+
+    def to_openeye(self) -> OEMol:
+        """OEChem representation of this molecule"""
+        return self.to_openff().to_openeye()
+
+    @classmethod
+    def from_openeye(cls, oemol: OEMol, name: str = ""):
+        raise NotImplementedError
+
+
+    def to_openff(self):
+        """OpenFF Toolkit representation of this molecule"""
+        m = OFFMolecule(self._rdkit, allow_undefined_stereo=True)
+        m.name = self.name
+
+        return m
+
+    @classmethod
+    def from_openff(cls, openff: OFFMolecule, name: str = ""):
+        """Construct from an OpenFF toolkit Molecule"""
+        return cls(openff.to_rdkit(), name=name)
+
+    def _to_dict(self) -> dict:
+        """Serialize to dict representation"""
+        # required attributes: (based on openff to_dict)
+        # for each atom:
+        #   element, name, formal charge, aromaticity, stereochemistry
+        # for each bond:
+        #   idx0, idx1, order, aromaticity, stereochemistry
+        # TODO: Do we care about fractional bond orders?
+        #       is aromaticity reperceived on creation?
+        # NOTE: Here we're implicitly using units of angstrom and elementary
+        # charge. We might want to explcitly include them in the stored dict.
+        m = self.to_openff()
+        atoms = [
+            (atom.atomic_number,
+             atom.name,
+             atom.formal_charge, #.m_as(unit.elementary_charge),
+             atom.is_aromatic,
+             atom.stereochemistry or '')
+            for atom in m.atoms
+        ]
+        bonds = [
+            (bond.atom1_index, bond.atom2_index, bond.bond_order,
+             bond.is_aromatic, bond.stereochemistry or '')
+            for bond in m.bonds
+        ]
+
+        if m.conformers is None:  # -no-cov-
+            # this should not be reachable; indicates that something went
+            # very wrong
+            raise RuntimeError(f"{self.__class__.__name__} must have at "
+                               "least 1 conformer")
+
+        conformers = [
+            serialize_numpy(conf) #.m_as(unit.angstrom)
+            for conf in m.conformers
+        ]
+
+        d = {
+            'atoms': atoms,
+            'bonds': bonds,
+            'name': self.name,
+            'conformers': conformers,
+        }
+
```

Comment:
so this looks like it needs the protein "layer" adding in.  One way for this is for `ProteinComponent` to have a `_to_dict` which calls `super()` and then adds in a few extra entries to the dict created here

off the top of my head, extra things include:
- resname, resid, icode per residue
- chain ids per chain
- grouping of atoms into residues (`[[0, 1, 2], [3, 4, 5], ...]`)
- grouping of residues into chains (`[[0, 1, 2, ....], [400, 401, ...]]`)

Reply:
should be there :)